Python/Product/TestAdapter.Executor/UnitTest/TestDiscovererUnitTest.cs

```diff
@@ -26,62 +58,90 @@ internal class TestDiscovererUnitTest : IPythonTestDiscoverer {
+            var workspaceText = _settings.IsWorkspace ? Strings.WorkspaceText : Strings.ProjectText;
+            LogInfo(Strings.PythonTestDiscovererStartedMessage.FormatUI(PythonConstants.UnitTestText, _settings.ProjectName, workspaceText, _settings.DiscoveryWaitTimeInSeconds));
 
             try {
                 var env = InitializeEnvironment(sources, _settings);
                 var arguments = GetArguments(sources);
-
-                using (var outputStream = new MemoryStream())
-                using (var writer = new StreamWriter(outputStream, Encoding.UTF8, 4096, leaveOpen: true))
-                using (var proc = ProcessOutput.Run(
-                    _settings.InterpreterPath,
+                json = ExecuteProcess(
+                    _settings.InterpreterPath, 
+                    env, 
                     arguments,
                     _settings.WorkingDirectory,
-                    env,
-                    visible: false,
-                    new StreamRedirector(writer)
-                )) {
-
-                    DebugInfo("cd " + _settings.WorkingDirectory);
-                    DebugInfo("set " + _settings.PathEnv + "=" + env[_settings.PathEnv]);
-                    DebugInfo(proc.Arguments);
-
-                    // If there's an error in the launcher script,
-                    // it will terminate without connecting back.
-                    WaitHandle.WaitAny(new WaitHandle[] { proc.WaitHandle });
-
-                    outputStream.Flush();
-                    outputStream.Seek(0, SeekOrigin.Begin);
-                    var json = new StreamReader(outputStream).ReadToEnd();
-
-                    try {
-                        unitTestResults = JsonConvert.DeserializeObject<List<UnitTestDiscoveryResults>>(json);
-                    } catch (InvalidOperationException ex) {
-                        Error("Failed to parse: {0}".FormatInvariant(ex.Message));
-                        Error(json);
-                    } catch (JsonException ex) {
-                        Error("Failed to parse: {0}".FormatInvariant(ex.Message));
-                        Error(json);
+                    _settings.PathEnv,
+                    _settings.DiscoveryWaitTimeInSeconds
+                    );
+            } catch (TimeoutException) {
+                Error(Strings.PythonTestDiscovererTimeoutErrorMessage);
+                return;
+            }
+
+            List<UnitTestDiscoveryResults> results = null;
+            try {
+                results = JsonConvert.DeserializeObject<List<UnitTestDiscoveryResults>>(json);
+            } catch (InvalidOperationException ex) {
+                Error("Failed to parse: {0}".FormatInvariant(ex.Message));
+                Error(json);
+            } catch (JsonException ex) {
+                Error("Failed to parse: {0}".FormatInvariant(ex.Message));
+                Error(json);
+            }
+
+            CreateVsTests(results, logger, discoverySink);
+        }
```

Comment:
this pretty much matches the TestDiscovereryPytest.cs now, just with different types